### PR TITLE
feat(quick-filter): Add show more for options.

### LINKED ISSUE #1169: [quick-filter] collapsable quick-filters OR a way to deal with filters with a huge set of values
# Feature Request

## Summary

We are using the quickfilter component and are quite happy with it for our general usecase. 
But we have a filter that has a lot of value-entries and makes it not very usable as a quickfilter.
(why we want this as a quickfilter see below) 
Now it would be awesome if there was a way to show a fixed size of filter values and "show more" if the user needs to see more. I dont really know if that is an ideal solution - I certainly wouldn't mind good suggestions in a different direction - but that is the one that I was thinking of. 

## Feature Description

- Problem 
We are using the quickfilter component for certain filters that are 
**1. used very often** -> that way the user needs just one click per filter  to filter the data instead of 3 clicks like they need in the filterfield 
(click into the filterfield; select the filterkey; select the filtervalue)
**2. very important for certain usecases** -> very important in the sense that the user sees on one glance all the values that are selectable.  
Now we have filters that have a huge set of filter-values (see screengrab for example; size of that value-list is ~500 entries)  This particular filter would be awesome to have as a quickfilter, because the filter itself is used very often. But the size of values prevents it from being useful as a quickfilter. 

- Solution
It would be awesome if there was some way to deal with this situation.

- one way could be using a "show more" -> where the user is given x (would need to be definable by the project using the component) entry values as a start and can click on "show more" which then presents with either the rest of the data or a given number of "more" entries. Similar to listing the contents of a huge directory on linux. (press Enter to show more - gives you more - pressing Enter again shows the next set of "more" - and so on) 
- another way could be to make the list collapsable 
- ... 

 I understand that my example is a rather extreme one (regarding the number of data entries) and maybe the quickfilter component is not the best solution for this entirely, but I thought I would put this in here so you know that there is demand for something like this. 

thx
 
## Attachments
image

### PR DESCRIPTION
**Pull request by lukasholzer**
Fixes #1169

### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by hubert-grininger** [comment]
Hi, I looked at the preview and I feel that this is not really a solution to our problem. First, the list of options will be potentially too long to show unlimited (especially for tags). Second, the workflow should work such that the "View more" jumps to the filter-field and automatically selects the specific filter. 

Please consider the current GWT implementation (eg. for the service list https://demo.dev.dynatracelabs.com/#newservices).
Thanks!

**Comment by hubert-grininger** [comment]
> Hi, I looked at the preview and I feel that this is not really a solution to our problem. First, the list of options will be potentially too long to show unlimited (especially for tags). Second, the workflow should work such that the "View more" jumps to the filter-field and automatically selects the specific filter.
> 
> Please consider the current GWT implementation (eg. for the service list https://demo.dev.dynatracelabs.com/#newservices).
> Thanks!

As discussed with Lukas, virtual scrolling should prevent problems with very long lists.

File: libs/barista-components/experimental/quick-filter/src/quick-filter.ts [hunk 1]
```diff
@@ -164,6 +182,12 @@ export class DtQuickFilter<T = any> implements AfterViewInit, OnDestroy {
    */
   @Input() groupHeadlineRole: number = 3;
 
+  /**
+   * The maximum amount of items that should be displayed in the quick filter
+   * sidebar. If there are more they are hided behind a show more functionality
```
Comment (rowa-audil) [inline libs/barista-components/experimental/quick-filter/src/quick-filter.ts:187]:
```suggestion
   * sidebar. If there are more they are hidden behind a show more functionality
```

**Review by SkaillZ — DISMISSED** [review]

File: libs/barista-components/experimental/quick-filter/README.md [hunk 2]
```diff
@@ -35,14 +35,16 @@ sidebar you can leverage the `<dt-quick-filter-title>` and
...
+| `aria-label`        | `string`                                            | `''`                               | Sets the value for the Aria-Label attribute.                                                                                                                                         |
+| `groupHeadlineRole` | `number`                                            | `3`                                | The aria-level of the group headlines for the document outline.                                                                                                                      |
+| `maxGroupItems`     | `number`                                            | `5`                                | The maximum amount of items that should be displayed in the quick filter sidebar. If there are more they are hided behind a show more functionality                                  |
+| `showMoreTemplate`  | `TemplateRef<{ $implicit: number; group: string }>` | `+145 options in the filter field` | Template for the show more text of the group. The implicit context of the template is the count of the remaining items and the view value can be accessed through the group variable |
```
[the remark this answers is not in the document]
Comment (rowa-audil) [reply libs/barista-components/experimental/quick-filter/README.md:47]: Revert that. `Count` is perfectly correct 😁

File: libs/barista-components/experimental/quick-filter/src/quick-filter-group.html [hunk 3]
```diff
@@ -6,35 +6,77 @@
...
+    (change)="_unsetGroup()"
   >
-    <dt-radio-button [checked]="_isNothingSelected()" (change)="_unsetGroup()">
-      Any
-    </dt-radio-button>
-    <dt-radio-button
-      *ngFor="let item of _getOptions()"
-      [value]="item"
-      [checked]="_isActive(item)"
+    Any
```
[the remark this answers is not in the document]
Comment (lukasholzer) [reply libs/barista-components/experimental/quick-filter/src/quick-filter-group.html:16]: Any is the value from the designs and should be used here

**Review by rowa-audil — APPROVED** [review]

**Review by rowa-audil — APPROVED** [review]
LGTM!

**Pull request merged**